operator/v1/types_dns.go

```diff
@@ -130,6 +131,68 @@ type Server struct {
 	ForwardPlugin ForwardPlugin `json:"forwardPlugin"`
 }
 
+// DNSTransport indicates what type of connection should be used
+// +kubebuilder:validation:Enum=TLS;Cleartext
+type DNSTransport string
+
+const (
+	// TLSTransport indicates that TLS should be used for the connection
```

Comment:
Nit, some of your comments end with a period and others don't, I don't have a strong opinion either way (though I lean to end with a period), but I do like consistency, so can we make sure they are consistent within themselves for this change